app/main.py

```diff
@@ -1,6 +1,47 @@
-def get_weather() -> None:
-    # write your code here
-    pass
+import os
+import requests
+from requests.exceptions import (
+    ConnectionError,
+    JSONDecodeError,
+    RequestException
+)
+
+
+URL = "http://api.weatherapi.com/v1/current.json?"
+API_KEY = os.environ["API_KEY"]
+FILTERING = os.environ.get("FILTERING", "Paris")
+
+
+def get_weather() -> dict:
+    url = f"{URL}q={FILTERING}&key={API_KEY}"
+    weather_data = fetch_data(url)
+    print_weather_data(weather_data)
+
+
+def fetch_data(url: str) -> dict:
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        data = response.json()
+        return data
+    except ConnectionError as exc:
+        print(f"Connection failed: {exc}")
+    except JSONDecodeError as exc:
+        print(f"Decoding failed: {exc}")
+    except RequestException as exc:
+        print(f"Request failed: {exc}")
+
+
+def print_weather_data(data: dict) -> None:
+    try:
+        print(
+            f'{data["location"]["name"]}/{data["location"]["country"]} '
+            f'{data["current"]["last_updated"]} '
+            f'Weather: {data["current"]["temp_c"]} '
+            f'Celsius, {data["current"]["condition"]["text"]}'
```

Comment:
The string formatting in `print_weather_data` uses double quotes within double quotes, which will cause a syntax error. Please correct this by using single quotes around the keys or escaping the double quotes.